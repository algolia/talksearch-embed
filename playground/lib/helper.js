import _ from 'lodash';
import pug from 'pug';
import fs from 'fs';
import path from 'path';
import $glob from 'glob';
import pify from 'pify';
const readFile = pify(fs.readFile);
const glob = pify($glob);

const helper = {
  glob,
  readFile,

  // Given a files list as given by metalsmith, will only return the path
  // matching the check method
  filterFiles(files, check) {
    return _.filter(_.keys(files), check);
  },

  pugToHtml(pugContent, data) {
    const pugOptions = {
      basedir: './playground/',
      ...data,
    };

    const content = pug.render(pugContent, pugOptions);

    return content;
  },

  // Find a file matching a specific pattern that is in the same folder and
  // returns its relative path
  async findSiblingFile(inputFile, pattern) {
    const dirname = _.get(inputFile, 'path.dir');
    const basedir = `./playground/src/${dirname}`;

    const matches = await glob(`${basedir}/${pattern}`);
    if (_.isEmpty(matches)) {
      return null;
    }
    const basename = path.basename(_.first(matches));
    return `./${basename}`;
  },
};

export default helper;
