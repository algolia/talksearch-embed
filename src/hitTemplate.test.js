/* eslint-disable import/no-commonjs */
import module from './hitTemplate.js';
import helper from './test-helper.js';

describe('hitTemplate', () => {
  describe('highlight', () => {
    it('should get the highlighted key', () => {
      const input = {
        _highlightResult: {
          foo: {
            value: 'bar',
          },
        },
      };

      const actual = module.internals.highlight(input, 'foo');

      expect(actual).toEqual('bar');
    });
    it('should default to the regular key if no highlight available', () => {
      const input = {
        foo: 'bar',
        _highlightResult: {
          notfoo: {
            value: 'baz',
          },
        },
      };

      const actual = module.internals.highlight(input, 'foo');

      expect(actual).toEqual('bar');
    });
  });

  describe('formatNumber', () => {
    it('should return the exact number if less than 1000', () => {
      const input = 123;

      const actual = module.internals.formatNumber(input);

      expect(actual).toEqual('123');
    });

    it('should return a rounded k-suffixed number if more than 1000', () => {
      const input = 1230;

      const actual = module.internals.formatNumber(input);

      expect(actual).toEqual('1k');
    });
  });

  describe('renderViews', () => {
    it('should return an element with 1k views', () => {
      const input = {
        video: {
          popularity: {
            views: 1234,
          },
        },
      };
      helper.mockPrivate(module, 'formatNumber', '1k');

      const actual = module.internals.renderViews(input);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('getThumbnail', () => {
    it('should return the highest quality thumbnail', () => {
      const input = {
        video: {
          thumbnails: {
            high: {
              url: 'foo',
            },
          },
        },
      };

      const actual = module.internals.getThumbnail(input);

      expect(actual).toEqual('foo');
    });
  });

  describe('renderCaption', () => {
    it('should not render anything if there is no match in the caption', () => {
      const input = {
        _snippetResult: {
          caption: {
            content: {
              matchLevel: 'none',
            },
          },
        },
      };

      const actual = module.internals.renderCaption(input);

      expect(actual).toEqual('');
    });

    it('should not render anything if there is no snippeting', () => {
      const input = {};

      const actual = module.internals.renderCaption(input);

      expect(actual).toEqual('');
    });

    it('should render a caption with the thumbnail as background', () => {
      const input = {
        _snippetResult: {
          caption: {
            content: {
              matchLevel: 'full',
              value: 'my_caption',
            },
          },
        },
      };
      helper.mockPrivate(module, 'getThumbnail', 'my_url');

      const actual = module.internals.renderCaption(input);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('renderSpeakers', () => {
    it('should return an empty string if no speakers', () => {
      const input = {};

      const actual = module.internals.renderSpeakers(input);

      expect(actual).toEqual('');
    });

    it('should render only one speaker', () => {
      const input = {
        speakers: [{ name: 'foo' }],
      };

      const actual = module.internals.renderSpeakers(input);

      expect(actual).toMatchSnapshot();
    });

    it('should render several speakers', () => {
      const input = {
        speakers: [{ name: 'foo' }, { name: 'bar' }],
      };

      const actual = module.internals.renderSpeakers(input);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('renderConferenceYear', () => {
    it('should render the year in an element', () => {
      const input = { conference: { year: 2018 } };

      const actual = module.internals.renderConferenceYear(input);

      expect(actual).toMatchSnapshot();
    });

    it('should render an empty string if no year found', () => {
      const input = {};

      const actual = module.internals.renderConferenceYear(input);

      expect(actual).toEqual('');
    });
  });

  describe('renderVideoSubtitle', () => {
    it('should only render the speaker if no year', () => {
      helper.mockPrivate(module, 'renderSpeakers', 'Tim Carry');
      helper.mockPrivate(module, 'renderConferenceYear', '');

      const actual = module.internals.renderVideoSubtitle();

      expect(actual).toMatchSnapshot();
    });

    it('should only render the year if no speakers', () => {
      helper.mockPrivate(module, 'renderSpeakers', '');
      helper.mockPrivate(module, 'renderConferenceYear', '2018');

      const actual = module.internals.renderVideoSubtitle();

      expect(actual).toMatchSnapshot();
    });
    it('should render both speakers and year', () => {
      helper.mockPrivate(module, 'renderSpeakers', 'Tim Carry');
      helper.mockPrivate(module, 'renderConferenceYear', '2018');

      const actual = module.internals.renderVideoSubtitle();

      expect(actual).toMatchSnapshot();
    });
  });
});
