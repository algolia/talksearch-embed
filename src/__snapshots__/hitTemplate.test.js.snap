// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hitTemplate renderCaption should render a caption with the thumbnail as background 1`] = `

<div class="ats-hit--captionBackdrop">
  <div class="ats-hit--captionBlur"
       style="background-image:url(my_url)"
  >
  </div>
</div>
<div class="ats-hit--caption">
  <div>
    my_caption
  </div>
</div>

`;

exports[`hitTemplate renderConferenceYear should render the year in an element 1`] = `

<span class="ats-hit--conferenceYear">
  2018
</span>

`;

exports[`hitTemplate renderSpeakers should render only one speaker 1`] = `

<span class="ats-hit--speakers">
  <span class="ats-hit--speakerName">
    foo
  </span>
</span>

`;

exports[`hitTemplate renderSpeakers should render several speakers 1`] = `

<span class="ats-hit--speakers">
  <span class="ats-hit--speakerName">
    foo
  </span>
  ,
  <span class="ats-hit--speakerName">
    bar
  </span>
</span>

`;

exports[`hitTemplate renderVideoSubtitle should only render the speaker if no year 1`] = `

<div class="ats-hit--videoSubtitle">
  Tim Carry
</div>

`;

exports[`hitTemplate renderVideoSubtitle should only render the year if no speakers 1`] = `

<div class="ats-hit--videoSubtitle">
  2018
</div>

`;

exports[`hitTemplate renderVideoSubtitle should render both speakers and year 1`] = `

<div class="ats-hit--videoSubtitle">
  Tim Carry in 2018
</div>

`;

exports[`hitTemplate renderViews should return an element with 1k views 1`] = `

<div class="ats-hit--views">
  1k views
</div>

`;
